src/promise.js

```diff
@@ -174,6 +174,11 @@ Promise.cast = function (obj) {
         ret._setFulfilled();
         ret._rejectionHandler0 = obj;
     }
+    if(debug.warnings() && typeof obj === "function"){
```

Comment:
> Checking for function should be done first.

Why? I figured debug.warnings() would be inlined as false so that'd be slightly faster. Changing just curious.